api/src/clojure/org/akvo/flow_api/datastore/survey.clj

```diff
@@ -1,16 +1,23 @@
 (ns org.akvo.flow-api.datastore.survey
   (:refer-clojure :exclude [list list*])
-  (:require [org.akvo.flow-api.anomaly :as anomaly]
-            [org.akvo.flow-api.datastore :as ds]
+  (:require [clojure.core.cache :as cache]
             clojure.set
-            [clojure.core.cache :as cache])
-  (:import [org.akvo.flow.api.dao FolderDAO SurveyDAO]
-           [com.google.appengine.api.datastore DatastoreService Entity Key KeyFactory QueryResultIterator]))
+            [org.akvo.flow-api.anomaly :as anomaly]
+            [org.akvo.flow-api.datastore :as ds])
+  (:import [com.gallatinsystems.survey.domain SurveyGroup]
+           [com.google.appengine.api.datastore DatastoreService Entity Key KeyFactory QueryResultIterator]
+           [org.akvo.flow.api.dao FolderDAO SurveyDAO]
+           [org.apache.commons.lang ArrayUtils]))
 
-(defn list* [user-id]
-  (let [survey-dao (SurveyDAO.)
-        all-surveys (.listAll survey-dao)]
-    (.filterByUserAuthorizationObjectId survey-dao all-surveys user-id)))
+(defn list*
+  ([user-id survey-ids]
```

Comment:
It is going to be very confusing if by mistake we pass a null here and end up getting all surveys. Same goes for the forms.

The duplication removal is not worth the possible confusion and overload of the survey-ids parameter meaning (null vs empty vs not-empty).

Reply:
> It is going to be very confusing if by mistake we pass a null here and end up getting all surveys. Same goes for the forms.

We **always filter** by authorization, the possibility of providing a set if ids (forms or surveys) is an _optimization_ to get fewer objects and and filter those by authorization. 

A call will never get _all surveys_ if the user is not authorized.

Does it make sense?